core/src/main/scala/org/apache/gearpump/cluster/Master.scala

```diff
@@ -78,6 +84,11 @@ private[cluster] class Master extends Actor with Stash {
       scheduler forward resourceUpdate
   }
 
+  def jarStoreService : Receive = {
```

Comment:
The JarContainer is created by Master and passed back to client, so that we can client can code transparently without knowing where the jar is stored. This is very flexible.

For example we may first use HDFS to store the jars, then HDFS cluster need a service upgrade, we cannot use it anymore, we can easily change this to use ftp to store it without upgrading client side binary.